### PR TITLE
DOCSUP-4524: Document the execute_merges_on_single_replica_time_threshold setting

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #16424

### CODE REVIEW

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -2459,4 +2458,19 @@ Possible values:
...
+Possible values:
+
+-   Positive integer (in seconds).
+-   0 — Special merges logic is not executed.
+
+Default value: `0`.
+
+**Special logic to perform merges**
+
+Selects one replica to perform the merge on. Sets the time threshold from the start of the merge. Other replicas wait for the merge to finish, then download the result. If the time threshold passes, and the selected replica does not perform merge, then the merge is performed manually.
```
Comment (filimonov) [inline docs/en/operations/settings/settings.md:2474]:
```suggestion
Selects one replica to perform the merge on. Sets the time threshold from the start of the merge. Other replicas wait for the merge to finish, then download the result. If the time threshold passes, and the selected replica does not perform merge, then the merge is performed on other replicas as usual. 

High values for that threshold may lead to replication delays. 
```

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -2459,4 +2458,19 @@ Possible values:
...
 Default value: `0`.
 
+## execute_merges_on_single_replica_time_threshold {#execute-merges-on-single-replica-time-threshold}
+
+Enables special logic to perform merges on replicas.
+
+Possible values:
+
+-   Positive integer (in seconds).
+-   0 — Special merges logic is not executed.
```
Comment (filimonov) [inline docs/en/operations/settings/settings.md:2468]:
```suggestion
-   0 — Special merges logic is not used. Merges happen in the  usual way on all the replicas.
```

File: docs/en/engines/table-engines/mergetree-family/replication.md [hunk 3]
```diff
@@ -113,6 +113,8 @@ You can have any number of replicas of the same data. Yandex.Metrica uses double
 
 The system monitors data synchronicity on replicas and is able to recover after a failure. Failover is automatic (for small differences in data) or semi-automatic (when data differs too much, which may indicate a configuration error).
 
+You can use a special logic for performing merges (for more information, see the documentation for [execute_merges_on_single_replica_time_threshold](../../../operations/settings/settings.md#execute-merges-on-single-replica-time-threshold) setting).
```
Comment (filimonov) [inline docs/en/engines/table-engines/mergetree-family/replication.md:116]: Not sure if we should advertise it too much :) It has sense only in some limited use cases when merges are CPU bounded not IO bounded.

File: docs/en/operations/settings/settings.md [hunk 4]
```diff
@@ -2459,4 +2458,19 @@ Possible values:
...
+
+Default value: `0`.
+
+**Special logic to perform merges**
+
```
Comment (filimonov) [inline docs/en/operations/settings/settings.md:2473]: Can be useful when merges are CPU bounded not IO bounded (heavy compression is in use, calculation expensive aggregate functions or default expressions or just very high number of tiny merges)

**Comment by sevirov** [comment]
@filimonov Please, review this PR.

**Comment by olgarev** [comment]
@filimonov Hi, this PR is waiting for your review.

**Comment by olgarev** [comment]
@qoega @alexey-milovidov Hi, please, assign smb to review this PR. It would be great to finish it today.

**Comment by filimonov** [comment]
Looking on it... (had short vacations)

**Review by filimonov — APPROVED** [review]

**Comment by olgarev** [comment]
@filimonov Thanks!

**Pull request merged**